.travis.yml

```diff
@@ -23,20 +23,20 @@ env:
     - ROS_DISTRO=hydro  USE_DEB=false      NOT_TEST_INSTALL=true
     - ROS_DISTRO=hydro  USE_DEB=true       NOT_TEST_INSTALL=true INSTALL_SRC="http://github.com/jsk-ros-pkg/jsk_pr2eus" TEST_PKGS="pr2eus"
     - ROS_DISTRO=hydro  USE_DEB=true       NOT_TEST_INSTALL=true BEFORE_SCRIPT="git clone -b hydro-devel http://github.com/ros/geometry; git clone -b hydro-devel http://github.com/ros/geometry-experimental; touch jsk_roseus/roseus_tutorials/CATKIN_IGNORE; touch geometry-experimental/test_tf2/CATKIN_IGNORE" # need to ignore roseus_tutorials which depends on image_view2 and that install deb version of tf/tf2
-    - ROS_DISTRO=hydro  USE_DEB=source
-    - ROS_DISTRO=indigo USE_DEB=true
-    - ROS_DISTRO=indigo USE_DEB=true       ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
-    - ROS_DISTRO=indigo USE_DEB=false      NOT_TEST_INSTALL=true
-    - ROS_DISTRO=indigo USE_DEB=source
-    - ROS_DISTRO=jade USE_DEB=true       TEST_PKGS="roseus roseus_smach" NOT_TEST_INSTALL=true
-    - ROS_DISTRO=jade USE_DEB=true       ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu TEST_PKGS="roseus roseus_smach" NOT_TEST_INSTALL=true
-    - ROS_DISTRO=jade USE_DEB=false      NOT_TEST_INSTALL=true TEST_PKGS="roseus roseus_smach"
-    - ROS_DISTRO=jade USE_DEB=source     TEST_PKGS="roseus roseus_smach" NOT_TEST_INSTALL=true
+    - ROS_DISTRO=hydro  USE_DEB=source     BEFORE_SCRIPT="sudo apt-get install -y xvfb"
+    - ROS_DISTRO=indigo USE_DEB=true       BEFORE_SCRIPT="sudo apt-get install -y xvfb"
+    - ROS_DISTRO=indigo USE_DEB=true       ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu BEFORE_SCRIPT="sudo apt-get install -y xvfb"
+    - ROS_DISTRO=indigo USE_DEB=false      NOT_TEST_INSTALL=true BEFORE_SCRIPT="sudo apt-get install -y xvfb"
+    - ROS_DISTRO=indigo USE_DEB=source   BEFORE_SCRIPT="sudo apt-get install -y xvfb"
```

Comment:
updated docker image on travis, now its ok

```
jenkins@jenkins:~/docker/ubuntu-14.04$ echo -e "FROM ros-ubuntu:14.04\nRUN
apt-get install -y xvfb\nEXPOSE 22" | sudo docker build -t ros-ubuntu:14.04
-
Sending build context to Docker daemon 2.048 kB
Sending build context to Docker daemon
Step 0 : FROM ros-ubuntu:14.04
 ---> 4118e57e601c
Step 1 : RUN apt-get install -y xvfb
 ---> Running in c26e869a0289
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following extra packages will be installed:
  libxfont1 libxkbfile1 x11-xkb-utils xfonts-base xfonts-encodings
  xfonts-utils xserver-common
Suggested packages:
  xfs xserver
The following NEW packages will be installed:
  libxfont1 libxkbfile1 x11-xkb-utils xfonts-base xfonts-encodings
  xfonts-utils xserver-common xvfb
0 upgraded, 8 newly installed, 0 to remove and 4 not upgraded.
Need to get 7936 kB of archives.
After this operation, 14.5 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu/ trusty-updates/main libxfont1 amd64
1:1.4.7-1ubuntu0.2 [95.0 kB]
Get:2 http://archive.ubuntu.com/ubuntu/ trusty/main libxkbfile1 amd64
1:1.0.8-1 [74.2 kB]
Get:3 http://archive.ubuntu.com/ubuntu/ trusty/main x11-xkb-utils amd64
7.7+1 [156 kB]
Get:4 http://archive.ubuntu.com/ubuntu/ trusty/main xfonts-encodings all
1:1.0.4-1ubuntu1 [583 kB]
Get:5 http://archive.ubuntu.com/ubuntu/ trusty/main xfonts-utils amd64
1:7.7+1 [73.9 kB]
Get:6 http://archive.ubuntu.com/ubuntu/ trusty/main xfonts-base all 1:1.0.3
[6180 kB]
Get:7 http://archive.ubuntu.com/ubuntu/ trusty-updates/main xserver-common
all 2:1.15.1-0ubuntu2.7 [27.9 kB]
Get:8 http://archive.ubuntu.com/ubuntu/ trusty-updates/main xvfb amd64
2:1.15.1-0ubuntu2.7 [747 kB]
Fetched 7936 kB in 35s (225 kB/s)

Selecting previously unselected package libxfont1:amd64.
(Reading database ... 93201 files and directories currently installed.)
Preparing to unpack .../libxfont1_1%3a1.4.7-1ubuntu0.2_amd64.deb ...
Unpacking libxfont1:amd64 (1:1.4.7-1ubuntu0.2) ...
Selecting previously unselected package libxkbfile1:amd64.
Preparing to unpack .../libxkbfile1_1%3a1.0.8-1_amd64.deb ...
Unpacking libxkbfile1:amd64 (1:1.0.8-1) ...
Selecting previously unselected package x11-xkb-utils.
Preparing to unpack .../x11-xkb-utils_7.7+1_amd64.deb ...
Unpacking x11-xkb-utils (7.7+1) ...
Selecting previously unselected package xfonts-encodings.
Preparing to unpack .../xfonts-encodings_1%3a1.0.4-1ubuntu1_all.deb ...
Unpacking xfonts-encodings (1:1.0.4-1ubuntu1) ...
Selecting previously unselected package xfonts-utils.
Preparing to unpack .../xfonts-utils_1%3a7.7+1_amd64.deb ...
Unpacking xfonts-utils (1:7.7+1) ...
Selecting previously unselected package xfonts-base.
Preparing to unpack .../xfonts-base_1%3a1.0.3_all.deb ...
Unpacking xfonts-base (1:1.0.3) ...
Selecting previously unselected package xserver-common.
Preparing to unpack .../xserver-common_2%3a1.15.1-0ubuntu2.7_all.deb ...
Unpacking xserver-common (2:1.15.1-0ubuntu2.7) ...
Selecting previously unselected package xvfb.
Preparing to unpack .../xvfb_2%3a1.15.1-0ubuntu2.7_amd64.deb ...
Unpacking xvfb (2:1.15.1-0ubuntu2.7) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Processing triggers for fontconfig (2.11.0-0ubuntu4.1) ...
Setting up libxfont1:amd64 (1:1.4.7-1ubuntu0.2) ...
Setting up libxkbfile1:amd64 (1:1.0.8-1) ...
Setting up x11-xkb-utils (7.7+1) ...
Setting up xfonts-encodings (1:1.0.4-1ubuntu1) ...
Setting up xfonts-utils (1:7.7+1) ...
Setting up xfonts-base (1:1.0.3) ...
Setting up xserver-common (2:1.15.1-0ubuntu2.7) ...
Setting up xvfb (2:1.15.1-0ubuntu2.7) ...
Processing triggers for libc-bin (2.19-0ubuntu6.6) ...
 ---> 6a86cd7ca27b
Removing intermediate container c26e869a0289
Step 2 : EXPOSE 22
 ---> Running in 6d0faf0e09c3
 ---> 7571fca3f3eb
Removing intermediate container 6d0faf0e09c3
Successfully built 7571fca3f3eb
jenkins@jenkins:~/docker/ubuntu-14.04$ sudo docker run --rm -i -t
ros-ubuntu:14.04 /bin/bash
root@dbfa1ca0be69:~# which xvfb
root@dbfa1ca0be69:~# aptitude show xvfb
Package: xvfb
State: installed
Automatically installed: no
Multi-Arch: foreign
Version: 2:1.15.1-0ubuntu2.7
Priority: optional
Section: x11
Maintainer: Ubuntu X-SWAT <ubuntu-x@lists.ubuntu.com>
Architecture: amd64
Uncompressed Size: 2191 k
Depends: xserver-common (>= 2:1.15.1-0ubuntu2.7), libaudit1 (>= 1:2.2.1),
libc6 (>= 2.17),
         libgcrypt11 (>= 1.5.1), libgl1-mesa-glx | libgl1, libpixman-1-0
(>= 0.30.0), libselinux1
         (>= 2.0.82), libxau6, libxdmcp6, libxfont1 (>= 1:1.4.2), xauth,
x11-xkb-utils
Recommends: libgl1-mesa-dri
Provides: xserver
Description: Virtual Framebuffer 'fake' X server
```

--
◉ Kei Okada


On Thu, Nov 12, 2015 at 12:53 PM, Furushchev <notifications@github.com>
wrote:

> In .travis.yml
> <https://github.com/jsk-ros-pkg/jsk_roseus/pull/393#discussion_r44619344>:
>
> > @@ -23,20 +23,20 @@ env:
> >      - ROS_DISTRO=hydro  USE_DEB=false      NOT_TEST_INSTALL=true
> >      - ROS_DISTRO=hydro  USE_DEB=true       NOT_TEST_INSTALL=true INSTALL_SRC="http://github.com/jsk-ros-pkg/jsk_pr2eus" TEST_PKGS="pr2eus"
> >      - ROS_DISTRO=hydro  USE_DEB=true       NOT_TEST_INSTALL=true BEFORE_SCRIPT="git clone -b hydro-devel http://github.com/ros/geometry; git clone -b hydro-devel http://github.com/ros/geometry-experimental; touch jsk_roseus/roseus_tutorials/CATKIN_IGNORE; touch geometry-experimental/test_tf2/CATKIN_IGNORE" # need to ignore roseus_tutorials which depends on image_view2 and that install deb version of tf/tf2
> > -    - ROS_DISTRO=hydro  USE_DEB=source
> > -    - ROS_DISTRO=indigo USE_DEB=true
> > -    - ROS_DISTRO=indigo USE_DEB=true       ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu
> > -    - ROS_DISTRO=indigo USE_DEB=false      NOT_TEST_INSTALL=true
> > -    - ROS_DISTRO=indigo USE_DEB=source
> > -    - ROS_DISTRO=jade USE_DEB=true       TEST_PKGS="roseus roseus_smach" NOT_TEST_INSTALL=true
> > -    - ROS_DISTRO=jade USE_DEB=true       ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu TEST_PKGS="roseus roseus_smach" NOT_TEST_INSTALL=true
> > -    - ROS_DISTRO=jade USE_DEB=false      NOT_TEST_INSTALL=true TEST_PKGS="roseus roseus_smach"
> > -    - ROS_DISTRO=jade USE_DEB=source     TEST_PKGS="roseus roseus_smach" NOT_TEST_INSTALL=true
> > +    - ROS_DISTRO=hydro  USE_DEB=source     BEFORE_SCRIPT="sudo apt-get install -y xvfb"
> > +    - ROS_DISTRO=indigo USE_DEB=true       BEFORE_SCRIPT="sudo apt-get install -y xvfb"
> > +    - ROS_DISTRO=indigo USE_DEB=true       ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu BEFORE_SCRIPT="sudo apt-get install -y xvfb"
> > +    - ROS_DISTRO=indigo USE_DEB=false      NOT_TEST_INSTALL=true BEFORE_SCRIPT="sudo apt-get install -y xvfb"
> > +    - ROS_DISTRO=indigo USE_DEB=source   BEFORE_SCRIPT="sudo apt-get install -y xvfb"
>
> I think No because commands on before_script was not not executed on
> jenkins.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_roseus/pull/393/files#r44619344>.
>